### PR TITLE
Add an optional dependency on jms/serializer-bundle

### PR DESCRIPTION
**Pull request by hason**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,10 +19,10 @@
         "symfony/framework-bundle": "~2.3",
         "symfony/assetic-bundle": "~2.1",
         "friendsofsymfony/rest-bundle": "~1.0",
-        "jms/serializer-bundle": "~0.12",
         "midgard/createphp": "~1.1"
     },
     "require-dev": {
+        "jms/serializer-bundle": "~0.12",
```
Comment (lsmith77) [inline composer.json]: it should then also be added to suggest .. would also be interesting to see if it also works with ``~1.0``

**Comment by lsmith77** [comment]
I guess we should then also add the core serializer as an optional dependency?

**Comment by dbu** [comment]
does the fos rest bundle now also work with the symfony serializer? do we actually directly depend on any serializer here or could we just point to the fos rest bundle?

**Comment by lsmith77** [comment]
the fos rest bundle does indeed work with the core serializer since many versions .. and yes .. we do not explicitly require any serializer .. however we do require that a serializer is configured in fos rest

**Comment by dbu** [comment]
hm, but is symfony/serializer not already required by depending on framework-bundle? if that is the case we need nothing more i think. if the user wants to tweak serialization and might need jms, he will have to look at FOSRestBundle anyways.

**Comment by dbu** [comment]
@lsmith77 what do you think about my last comment?

**Comment by lsmith77** [comment]
@dbu the serializer is not required by the framework bundle .. I just submitted a PR to add it as a suggest.

**Comment by dbu** [comment]
okay. @hason can you please update the PR to also suggest the symfony serializer as other alternative? and can you check if it works with jms serializer 1.0 as well? FOSRest is doing this: https://github.com/FriendsOfSymfony/FOSRestBundle/blob/1.7/composer.json#L43-L44

**Comment by hason** [comment]
@dbu updated

**Comment by dbu** [comment]
@lsmith77 can you look at this please?

File: composer.json [hunk 2]
```diff
@@ -19,13 +18,16 @@
...
         "midgard/createphp": "~1.1"
     },
     "require-dev": {
+        "symfony/serializer": "~2.3",
+        "jms/serializer-bundle": "~0.12|~1.0",
         "symfony-cmf/testing": "~1.2,>=1.2.6"
     },
     "suggest": {
+        "jms/serializer-bundle": "Add support for advanced serialization capabilities, recommended, requires ~0.12|~1.0",
+        "symfony/serializer": "Add support for basic serialization capabilities and xml decoding, requires ~2.3",
```
Comment (dbu) [inline composer.json]: i think we should mention FOSRestBundle here, as we do not interact with the serializer directly.

**Comment by dbu** [comment]
we need to update the doc to specify that you need to require a serializer if your project does not use symfony/symfony which includes the symfony/serializer. the doc is at https://github.com/symfony-cmf/symfony-cmf-docs/blob/master/bundles/create/introduction.rst - the setup instructions say to instantiate jms serializer bundle, which is now only correct if you do use it...

i am a bit worried with the developer experience. what happens if the dev did not notice he needs a serializer? i think right now we will get some confusing problem with FOSRest at runtime. i suggest that in the [DI extension class](https://github.com/symfony-cmf/CreateBundle/blob/master/DependencyInjection/CmfCreateExtension.php), we check for the existence of either the main class of symfony/serializer or the jms serializer being loaded, and if not present throw an exception explaining the problem to the dev.

**Comment by lsmith77** [comment]
I agree @dbu .. we should handle this in the DI extension class

**Comment by dbu** [comment]
@hason would you be ok to change this PR to have the DI extension check the bundles for one of the two serializers being present?

**Comment by hason** [comment]
@dbu, @lsmith77 The test for the presence of a serializer is made in FOSRestBundle https://github.com/FriendsOfSymfony/FOSRestBundle/blob/master/DependencyInjection/Compiler/SerializerConfigurationPass.php#L32

**Comment by dbu** [comment]
ah, indeed. i think with that we are good already, wdyt @lsmith77?

File: composer.json [hunk 3]
```diff
@@ -13,19 +13,19 @@
             "homepage": "https://github.com/symfony-cmf/CreateBundle/contributors"
         }
     ],
-    "minimum-stability": "dev",
```
Comment (lsmith77) [inline composer.json]: please leave this unchanged ..

Comment (hason) [inline composer.json]: Fixed

**Comment by dbu** [comment]
thanks!

**Pull request merged**